usr/share/rear/layout/prepare/Linux-s390/205_s390_enable_disk.sh

```diff
@@ -2,45 +2,36 @@
 # Before we can compare or map DASD devices we must enable them.
 # This operation is only needed during "rear recover".
 
-format_s390_disk() {
-    LogPrint "run dasdfmt"
-    while read line ; do
-        LogPrint 'dasdfmt:' "$line"
-        # example format command: dasdfmt -b 4096 -d cdl -y /dev/dasda
-        # where
-        #  b is the block size
-        #  d is the layout: 
-        #   cdl - compatible disk layout (can be shared with zos and zvm apps)
-        #   ldl - linux disk layout
-        #  y - answer yes
-        device=$( echo $line | awk '{ print $7 }' )
-        blocksize=$( echo $line | awk '{ print $3 }' )
-        layout=$( echo $line | awk '{ print tolower($5) }' )
-        if [[ "$layout" == "ldl" ]] ; then
-            # listDasdLdl contains devices such as /dev/dasdb that are formatted as LDL
-            # LDL formatted disks are already partitioned and should not be partitioned with parted or fdasd , it will fail
-            # this var, listDasdLdl, is used by 100_include_partition_code.sh to exclude writing partition code to diskrestore.sh for LDL disks
-            listDasdLdl+=( $device )
-            LogPrint "LDL disk '$device' will not be partitioned (LDL disks are already partitioned)"
-        fi
-        LogPrint 'dasdfmt:' $device ', blocksize:' $blocksize ', layout:' $layout
-        # dasd format
-        dasdfmt -b $blocksize -d $layout -y $device
-    done < <( grep "^dasdfmt " "$LAYOUT_FILE" )
-}
-
+DISK_MAPPING_HINTS=()
 
 enable_s390_disk() {
+    local keyword device bus len newname
+
     LogPrint "run chccwdev"
-    while read line ; do
-        LogPrint 'dasd channel:' "$line"
-        device=$( echo $line | awk '{ print $4 }' )
-        bus=$( echo $line | awk '{ print $2 }' )
-        channel=$( echo $line | awk '{ print $5 }' )
-        LogPrint 'chccwdev:' $device ', bus:' $bus ', channel:' $channel
-        # dasd channel enable
-        chccwdev -e $bus
-    done < <( grep "^dasd_channel " "$LAYOUT_FILE" )
+    while read len device bus ; do
+        # this while loop must be outside the pipeline so that variables propagate outside
+        # (pipelines run in subshells)
+        LogPrint "Enabling DASD $device with virtual device number $bus"
+        if chccwdev -e $bus ; then
+            newname=$(lsdasd $bus | awk "/$bus/ { print \$3}" )
```

Comment:
side note: all those s390 specific tools are added in `usr/share/rear/prep/Linux-s390/305_include_s390_tools.sh` to `PROGS` - shouldn't that be rather `REQUIRED_PROGS` to ensure that the recovery code won't fail due to missing tools?

Reply:
It should. I noticed that as well and intended to submit a separate PR.